### ISSUE TITLE
[BUG] QueueClient.ReceiveMessagesAsync throws System.Xml.XmlException

### ISSUE THREAD

**Issue by Alanddq** [Azure/azure-sdk-for-net#19501]
**Describe the bug**
QueueClient.ReceiveMessagesAsync throws XmlException. From SDK client perspective, it is not expected behavior.

**Expected behavior**
QueueClient.ReceiveMessagesAsync should either succeed or failed with RequestFailedException which indicates the error from queue service.

**Actual behavior (include Exception or Stack Trace)**
QueueClient.ReceiveMessagesAsync throws XmlException. Looks like SDK get a response from queue service but failed to parse in Xml.

```text
-------Exception information------
System.Xml.XmlException: '"' is an unexpected token. Expecting white space. Line 1, position 50.
at System.Xml.XmlTextReaderImpl.Throw(Exception e)
at System.Xml.XmlTextReaderImpl.DtdParserProxy.System.Xml.IDtdParserAdapter.Throw(Exception e)
at System.Xml.DtdParser.ParseExternalId(Token idTokenType, Token declType, String& publicId, String& systemId)
at System.Xml.DtdParser.ParseInDocumentDtd(Boolean saveInternalSubset)
at System.Xml.DtdParser.Parse(Boolean saveInternalSubset)
at System.Xml.DtdParser.System.Xml.IDtdParser.ParseInternalDtd(IDtdParserAdapter adapter, Boolean saveInternalSubset)
at System.Xml.XmlTextReaderImpl.ParseDtd()
at System.Xml.XmlTextReaderImpl.ParseDoctypeDecl()
at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
at System.Xml.Linq.XDocument.Load(Stream stream, LoadOptions options)
at Azure.Storage.Queues.QueueRestClient.Messages.DequeueAsync_CreateResponse(ClientDiagnostics clientDiagnostics, Response response)
at Azure.Storage.Queues.QueueRestClient.Messages.<DequeueAsync>d__0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Azure.Storage.Queues.QueueClient.<ReceiveMessagesInternal>d__90.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Azure.Storage.Queues.QueueClient.<ReceiveMessagesAsync>d__89.MoveNext()
```

**To Reproduce**
This issue doesn't happen frequently, around once per month in our production VMs which polling messages from queue. We cannot reproduce it on purpose.

Something interesting to mention is this issue is not persistent. It only happened once and then ReceiveMessagesAsync can succeed after. So I don’t think it is caused by invalid message content, otherwise it should keep happening since we don’t delete message if failed to get it.

**Environment:**
 - Name and version of the Library package used: Azure.Storage.Queues 12.5.0]
 - Hosting platform or OS and .NET runtime version (`dotnet --info` output for .NET Core projects): Windows 10 .NET Framework 4.7
 - IDE and version :

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team best able to assist.

**Comment by seanmcc-msft** [comment]
@jaschrep-msft, weren't you working on this, or a similar issue?

**Comment by jaschrep-msft** [comment]
I don't recall working on an issue like this.

@Alanddq there really isn't enough here to go off of. Is it possible for you to log the returned XML when this exception occurs and share with us? Sanitize your private data from the messages, of course. Considering whitespace is expected, the issue should be in the tags themselves and not their contents.

Position 50 is a weird spot to get this, though; it's in the middle of the `<QueueMessagesList>` tag and the schema doesn't provide any variation that early in the response. Something else has to be occurring.

**Comment by Alanddq** [comment]
That's everything we can provide at this point. It happened on our production VM, and we don't log returned XML there.

We also disabled SDK log due to a memory leak issue here: https://github.com/Azure/azure-sdk-for-net/issues/18199. So we cannot provide log either.

As I mentioned in the description, this looks like a transient issue. ReceiveMessagesAsync can succeed after. Hope this information can provide some hint for investigation. @jaschrep-msft

**Comment by kasobol-msft** [comment]
Related https://github.com/Azure/azure-sdk-for-net/issues/10294

**Comment by kasobol-msft** [comment]
This has been addressed in https://github.com/Azure/azure-sdk-for-net/pull/17786 and is available in `12.7.0-beta.1` version.

Before the error handling logic attempted to parse xml on error response without checking for content type first , see https://github.com/Azure/azure-sdk-for-net/blob/52fb9b7628732e70a41dd057625619dcbaf31479/sdk/storage/Azure.Storage.Queues/src/Generated/QueueRestClient.cs#L1700-L1704
In latest version this has been fixed, see https://github.com/Azure/azure-sdk-for-net/blob/a88d63557f88bbf1162ef09bd242953e1d536d06/sdk/storage/Azure.Storage.Common/src/Shared/ClientDiagnostics.cs#L37-L91
